charts/aws-vpc-cni/templates/daemonset.yaml

```diff
@@ -0,0 +1,158 @@
+      {{- end }}
+      labels:
+        app.kubernetes.io/name: {{ include "aws-vpc-cni.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        k8s-app: aws-node
+    spec:
+      priorityClassName: "{{ .Values.priorityClassName }}"
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: "beta.kubernetes.io/os"
+                    operator: In
+                    values:
+                      - linux
+                  - key: "beta.kubernetes.io/arch"
+                    operator: In
+                    values:
+                      - amd64
+                      - arm64
+                  - key: "eks.amazonaws.com/compute-type"
+                    operator: NotIn
+                    values:
+                      - fargate
+              - matchExpressions:
+                  - key: "kubernetes.io/os"
+                    operator: In
+                    values:
+                      - linux
+                  - key: "kubernetes.io/arch"
+                    operator: In
+                    values:
+                      - amd64
+                      - arm64
+                  - key: "eks.amazonaws.com/compute-type"
+                    operator: NotIn
+                    values:
+                      - fargate
+      serviceAccountName: {{ template "aws-vpc-cni.serviceAccountName" . }}
+      hostNetwork: true
+      initContainers:
+      - name: aws-vpc-cni-init
+        image: "{{- if .Values.initContainers.image.override }}{{- .Values.initContainers.image.override }}{{- else }}{{- .Values.initContainers.image.account }}.dkr.ecr.{{- .Values.initContainers.image.region }}.{{- .Values.initContainers.image.domain }}/amazon-k8s-cni-init:{{- .Values.initContainers.image.tag }}{{- end}}"
+        imagePullPolicy: {{ .Values.initContainers.image.pullPolicy }}
+        env:
+{{- range $key, $value := .Values.initContainers.env }}
+          - name: {{ $key }}
+            value: {{ $value | quote }}
+{{- end }}
+        securityContext:
+          {{- toYaml .Values.initContainers.securityContext | nindent 12 }}
+        volumeMounts:
+          - mountPath: /host/opt/cni/bin
+            name: cni-bin-dir
+      terminationGracePeriodSeconds: 10
+      tolerations:
+        - operator: Exists
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: aws-node
+          image: "{{- if .Values.image.override }}{{- .Values.image.override }}{{- else }}{{- .Values.image.account }}.dkr.ecr.{{- .Values.image.region }}.{{- .Values.image.domain }}/amazon-k8s-cni:{{- .Values.image.tag }}{{- end}}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - containerPort: 61678
```

Comment:
should this always be static or should it be configurable?

Reply:
Will make it configurable too.